blueprints/in-repo-engine/files/lib/__name__/index.js

```diff
@@ -0,0 +1,8 @@
+/*jshint node:true*/
+module.exports = {
+  name: '<%= dasherizedModuleName %>',
+
+  isDevelopingAddon: function() {
+    return true;
+  }
+};
```

Comment:
@rwjblue from my experiments, it seems like the hinting errors only get thrown if you are using an actual addon (or symlinking an in-repo-addon as is done in the ember-engines tests). When using a straight up in-repo-addon as generated by default, it works without any issues.